app/src/main/java/com/gusta/wakemehome/utilities/PermissionUtils.java

```diff
@@ -0,0 +1,117 @@
+package com.gusta.wakemehome.utilities;
+
+import android.Manifest;
+import android.app.Activity;
+import android.content.Context;
+import android.content.Intent;
+import android.content.pm.PackageManager;
+import android.net.Uri;
+import android.provider.Settings;
+import android.util.Log;
+import android.view.View;
+
+import androidx.annotation.NonNull;
+import androidx.core.app.ActivityCompat;
+
+import com.gusta.wakemehome.BuildConfig;
+import com.gusta.wakemehome.R;
+import com.gusta.wakemehome.geofencing.GeofenceManager;
+
+public final class PermissionUtils {
+
+    // Constant for logging
+    private static final String TAG = GeofenceManager.class.getSimpleName();
+
+    /**
+     * Return the current state of the permissions needed.
+     */
+    public static boolean missingPermissions(Context context) {
+        int permissionState = ActivityCompat.checkSelfPermission(context,
+                Manifest.permission.ACCESS_FINE_LOCATION);
+        return permissionState != PackageManager.PERMISSION_GRANTED;
+    }
+
+    /**
+     * Ask the user for permission.
+     */
+    public static void requestPermissions(final Activity activity) {
+
+        // Request permissions from the user
+        boolean shouldProvideRationale =
+                ActivityCompat.shouldShowRequestPermissionRationale(activity,
```

Comment:
Why did you add it to specific permission ?
it's easy to change it to generic with permission name?

Reply:
You are right I will change.
I thought it would be problematic for the "settings" link in the snackbar, but it won't (all permissions should send user to "app details" setting).